### ISSUE TITLE
Set up CI

### ISSUE THREAD

**Issue by jlndnn** [open-source-siempo/siempo-android-launcher#2]
Once we have figured out the local build process, we need to convert this to CI. As part of our org plan we get some free Github actions minutes per month - might be worth looking into.

**Comment by gdryke** [comment]
I think I've got some basics going here, at least the build! No tests yet really, right?

https://github.com/gdryke/focus-launcher-android/runs/2083445901?check_suite_focus=true

**Comment by jlndnn** [comment]
Yes that's right, I haven't seen any tests so far. Build works fine for me locally too, will take a look at your CI yml, seems fine apart from the gradle step so hopefully it's something small 🤔

**Comment by zed13** [comment]
I propose new scheme for releases. There is three num version X.Y.Z:
1. X - major releases for some significant changes
2. Y - one or few feature release with bugfixes.
3. Z - bugfix or hotfix.

There is different ways of branching. But I propose the following. 

Use three kinds of branches:
1. master branch - here we collect feature PRs and bugfixes for next release
2. hotfix branch - is used when it's required to fix and publish something without release
3. feature/bugfix branch - it used for every feature and bugfix. could be nested

At start of each release new version is set up in gradle build script and with PR (or without) master change it's version. In general it will looks like 1.0.10 -> 1.1.0. 
All following PR's with new features and bugfixes should be targeted to master. When all required PR's merget to master we starts stabilization of master - testing and bugfixing. 
When master is stabilized somebody marks last commit in master with tag release_X.Y.Z and builds artifact for Google Play.

When some critical bugs leaked to user we could make hotfix. Create hotfix branch from release tag of active release.  Make required fixes. Test them. Publish hotfix to Google play. And don't forget to merge hotfix changes to master.
